client/apis/volumegroupsnapshot/v1beta1/types.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 // +kubebuilder:object:generate=true
-package v1alpha1
```

Comment:
Why is `VolumeGroupSnapshot.Status.CreationTime` --> `metav1.Time` and why is `VolumeGroupSnapshotContentStatus.CreationTime` --> int64 ?

I know that change was not introduced in this PR, but since we are moving this to beta, we have to be careful about all the decisions we make.

Reply:
The same is happening with independent VolumeSnapshots:

https://github.com/kubernetes-csi/external-snapshotter/blob/07425bd341a786aa58eecb1d2dcec81b5e6dcfe8/client/apis/volumesnapshot/v1/types.go#L158

vs:

https://github.com/kubernetes-csi/external-snapshotter/blob/07425bd341a786aa58eecb1d2dcec81b5e6dcfe8/client/apis/volumesnapshot/v1/types.go#L408 

The content object looks more directly related to the GRPC CSI call.

Reply:
This is how it is displayed in CLI then:

```
  creationTime: "2024-11-22T22:14:35Z"
  creationTime: 1732313675689050917
```

@xing-yang can you confirm if this intentional? This doesn't seem right.

Reply:
We can always convert `int64` to `metav1.Time` so as this is nicely handled by k8s tooling. I am not sure if this has to be int64...

Reply:
In CSI spec, the timestamp is .google.protobuf.Timestamp which represents seconds of UTC time since Unix epoch 1970-01-01T00:00:00Z. So in VolumeSnapshotContent and VolumeGroupSnapshotContent, it is a direct mapping to what is returned from the CSI driver.
VolumeSnapshot and VolumeGroupSnapshot are in user namespace, we try to show them nicely in the readable format of `metav1.Time`.

Reply:
I still do not understand why though. `metav1.Time` also represents timestamp equally well. In k8s APIs, `metav1.Time` is almost universally used for representing time. Is there a loss of information or something we will incur if we use `metav1.Time`?

Reply:
VolumeSnapshotContent and VolumeGroupSnapshotContent are admin API objects. We don't need to show them in user readable format.  I think they should stay true to the original values returned by the CSI driver.
Since we've already made a decision for this in VolumeSnapshotContent, it also looks inconsistent now if we are showing them differently in VolumeGroupSnapshotContent.

Reply:
Since VolumeSnapshot and VolumeGroupSnapshot already have timestamps shown in user readable format, my preference is to keep the timestamps in VolumeSnapshotContent and VolumeGroupSnapshotContent the same as what is returned by the CSI driver as they are admin facing, but let's see what @jsafrane and @msau42 think.

Reply:
@leonardoce can you please update the comments with the highlighted text?

type VolumeGroupSnapshotStatus struct {
..........
	// CreationTime is the timestamp when the point-in-time group snapshot is taken
	// by the underlying storage system.
	// If not specified, it may indicate that the creation time of the group snapshot
	// is unknown.
	// The format of this field is a Unix nanoseconds time encoded as an int64.
	// On Unix, the command date +%s%N returns the current time in nanoseconds
	// since 1970-01-01 00:00:00 UTC.
        **// This field is updated based on the CreationTime field in VolumeGroupSnapshotContentStatus**
	// +optional
	CreationTime *metav1.Time `json:"creationTime,omitempty" protobuf:"bytes,2,opt,name=creationTime"`

type VolumeGroupSnapshotContentStatus struct {
..........
	// CreationTime is the timestamp when the point-in-time group snapshot is taken
	// by the underlying storage system.
	// If not specified, it indicates the creation time is unknown.
	// If not specified, it means the readiness of a group snapshot is unknown.
	// The format of this field is a Unix nanoseconds time encoded as an int64.
	// On Unix, the command date +%s%N returns the current time in nanoseconds
	// since 1970-01-01 00:00:00 UTC.
        **// This field is the source for the CreationTime field in VolumeGroupSnapshotStatus**
	// +optional
	CreationTime *int64 `json:"creationTime,omitempty" protobuf:"varint,2,opt,name=creationTime"`

Reply:
I improved the comments @xing-yang.

Reply:
I believe we concluded last week to use `metav1.Time` for CreationTime?

Reply:
Opened an issue: https://github.com/kubernetes-csi/external-snapshotter/issues/1234
cc @leonardoce

Reply:
@msau42 PR is submitted here: https://github.com/kubernetes-csi/external-snapshotter/pull/1235